cmd/endpoint.go

```diff
@@ -190,18 +184,13 @@ func (endpoints EndpointList) Less(i, j int) bool {
 	return endpoints[i].String() < endpoints[j].String()
 }
 
-// SetHTTPS - sets secure http for URLEndpointType.
-func (endpoints EndpointList) SetHTTPS() {
-	for i := range endpoints {
-		endpoints[i].SetHTTPS()
-	}
-}
-
-// SetHTTP - sets insecure http for URLEndpointType.
-func (endpoints EndpointList) SetHTTP() {
+// IsHTTPS - returns true if secure for URLEndpointType.
+func (endpoints EndpointList) IsHTTPS() bool {
+	var ok = true
 	for i := range endpoints {
-		endpoints[i].SetHTTP()
+		ok = ok && endpoints[i].IsHTTPS()
 	}
+	return ok
```

Comment:
easier would be:
```
{
for i := range endpoints {
    if !endpoints[i].IsHTTPS() {
        return false
    }
return true
}
```

Reply:
How is this easier ?

Reply:
Second function is easier to read and also more efficient (since we return earlier):
```
func (endpoints EndpointList) IsHTTPS() bool {
        var ok = true
        for i := range endpoints {
                ok = ok && endpoints[i].IsHTTPS()
        }
        return ok
}
```

```
func (endpoints EndpointList) IsHTTPS() bool {
        for _, endpoint := range endpoints {
                if endpoint.IsHTTPS() == false {
                        return false
                }
        }
        return true
}
```